Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -1036,6 +1279,52 @@ Optional<double> writable_stream_default_writer_get_desired_size(WritableStreamD
     return writable_stream_default_controller_get_desired_size(*stream->controller());
 }
 
+// https://streams.spec.whatwg.org/#writable-stream-default-controller-advance-queue-if-needed
+WebIDL::ExceptionOr<void> writable_stream_default_controller_advance_queue_if_needed(WritableStreamDefaultController& controller)
+{
+    // 1. Let stream be controller.[[stream]].
+    auto stream = controller.stream();
+
+    // 2. If controller.[[started]] is false, return.
+    if (!controller.started())
+        return {};
+
+    // 3. If stream.[[inFlightWriteRequest]] is not undefined, return.
+    if (stream->in_flight_write_request())
+        return {};
+
+    // 4. Let state be stream.[[state]].
+    auto state = stream->state();
+
+    // 5. Assert: state is not "closed" or "errored".
+    VERIFY(state != WritableStream::State::Closed && state != WritableStream::State::Errored);
+
+    // 6. If state is "erroring",
+    if (state == WritableStream::State::Erroring) {
+        // 1. Perform ! WritableStreamFinishErroring(stream).
+        // 2. Return.
+        return writable_stream_finish_erroring(*stream);
+    }
+
+    // 7. If controller.[[queue]] is empty, return.
+    if (controller.queue().is_empty())
+        return {};
+
+    // 8. Let value be ! PeekQueueValue(controller).
+    auto value = peek_queue_value(controller);
+
+    // 9. If value is the close sentinel, perform ! WritableStreamDefaultControllerProcessClose(controller).
+    if (is_close_sentinel(value)) {
+        TRY(writable_stream_default_controller_process_close(controller));
+    }
+    else {
+        // 10. Otherwise, perform ! WritableStreamDefaultControllerProcessWrite(controller, value).
```

Comment:
Nit: `Otherwise` comment should go before the `else`, even if it includes the actual instruction what to do.

Reply:
ofc, went a little too fast with the fixups!